dglgo/dglgo/model/graph_model/pna.py

```diff
@@ -0,0 +1,230 @@
+        h_new = F.relu(h_new)
+
+        if self.residual:
+            h_new = h_new + h
+        h_new = self.dropout(h_new)
+
+        return h_new
+
+class PNA(nn.Module):
+    def __init__(self,
+                 data_info: dict,
+                 embed_size: int = 80,
+                 aggregators: str = 'mean max min std',
+                 scalers: str = 'identity amplification attenuation',
+                 dropout: float = 0.3,
+                 batch_norm: bool = True,
+                 residual: bool = True,
+                 num_mlp_layers: int = 1,
+                 num_layers: int = 4,
+                 readout: str = 'mean'):
+        """Principal Neighbourhood Aggregation
+
+        Parameters
+        ----------
+        data_info : dict
+            The information about the input dataset.
+        embed_size : int
+            Embedding size.
+        aggregators : str
+            Aggregation function names separated by space, can include mean, max, min, std, sum
+        scalers : str
+            Scaler function names separated by space, can include identity, amplification, and attenuation
+        dropout : float
+            Dropout rate.
+        batch_norm : bool
+            Whether to use batch normalization.
+        residual : bool
+            Whether to use residual connection.
+        num_mlp_layers : int
+            Number of MLP layers to use after message aggregation in each PNA layer.
+        num_layers : int
+            Number of PNA layers.
+        readout : str
+            Readout for computing graph-level representations, can be 'sum' or 'mean'.
+        """
+        super(PNA, self).__init__()
+        self.data_info = data_info
+        self.embed_size = embed_size
+        self.dropout = dropout
+        self.batch_norm = batch_norm
+        self.residual = residual
+        self.num_mlp_layers = num_mlp_layers
+        self.num_layers = num_layers
+        self.readout = readout
+
+        if aggregators is None:
+            aggregators = ['mean', 'max', 'min', 'std']
+        else:
+            aggregators = [agg.strip() for agg in aggregators.split(' ')]
+            assert set(aggregators).issubset({'mean', 'max', 'min', 'std', 'sum'}), \
+                "Expect aggregators to be a subset of ['mean', 'max', 'min', 'std', 'sum'], \
+                    got {}".format(aggregators)
+        if scalers is None:
+            scalers = ['identity', 'amplification', 'attenuation']
+        else:
+            scalers = [scl.strip() for scl in scalers.split(' ')]
+            assert set(scalers).issubset({'identity', 'amplification', 'attenuation'}), \
+                "Expect scalers to be a subset of ['identity', 'amplification', 'attenuation'], \
+                    got {}".format(scalers)
+        self.aggregators = aggregators
+        self.scalers = scalers
+
+        if data_info['name'] in ['ogbg-molhiv', 'ogbg-molpcba']:
+            self.node_encoder = AtomEncoder(embed_size)
```

Comment:
Expose the choice of `node_encoder` and `edge_encoder` as model hyperparameter?

Reply:
What will be the possible options?

Reply:
For example, we can expose `node_encoder` as a string with two options `'atom'` for `AtomEncoder` while `'linear'` for `nn.Linear`.